crates/interpreter/src/instructions/opcode.rs

```diff
@@ -28,11 +28,55 @@ pub type BoxedInstructionTable<'a, H> = [BoxedInstruction<'a, H>; 256];
 /// Note that `Plain` variant gives us 10-20% faster Interpreter execution.
 ///
 /// Boxed variant can be used to wrap plain function pointer with closure.
-pub enum InstructionTables<'a, H: Host> {
+pub enum InstructionTables<'a, H> {
     Plain(InstructionTable<H>),
     Boxed(BoxedInstructionTable<'a, H>),
 }
 
+impl<H: Host> InstructionTables<'_, H> {
+    /// Creates a plain instruction table for the given spec.
+    #[inline]
+    pub const fn new_plain<SPEC: Spec>() -> Self {
+        Self::Plain(make_instruction_table::<H, SPEC>())
+    }
+}
+
+/// Make instruction table.
+#[inline]
+pub const fn make_instruction_table<H: Host, SPEC: Spec>() -> InstructionTable<H> {
+    // Force const-eval of the table creation, making this function trivial.
+    // TODO: Replace this with a `const {}` block once it is stable.
+    struct ConstTable<H: Host, SPEC: Spec> {
+        _phantom: core::marker::PhantomData<(H, SPEC)>,
+    }
+    impl<H: Host, SPEC: Spec> ConstTable<H, SPEC> {
+        const NEW: InstructionTable<H> = {
+            let mut tables: InstructionTable<H> = [control::unknown; 256];
+            let mut i = 0;
+            while i < 256 {
+                tables[i] = instruction::<H, SPEC>(i as u8);
+                i += 1;
+            }
+            tables
+        };
+    }
```

Comment:
Didn't know this was possible. Nice one!

llvm got fixed in 1.75 so we could use newest rust.
But will merge this as it is.